File: pset6/mario/more/mario.py
# ----------------------------------------------
# CS50 Problem Set 6: Sentimental - Mario (more)
#
# Summary: Reimplements Mario (more) in python
# Prints Mario`s pyramid (more)
#
# Lucas Emidio Fernandes Dias
# 03 October 2018
# ----------------------------------------------

from cs50 import get_int

# Constants
# Gap width
GAP = 2

# Asks the user for a height until height is valid
while True:

    height = get_int("Height: ")

    # Height is only valid if 0 <= height <= 23
    if height >= 0 and height <= 23:
        break

# Prints the pyramid
for i in range(height):

    # Print left pyramid
    # Print spaces
    print(" " * (height - 1 - i), end="")

    # Print hashes
    print("#" * (i + 1), end="")

    # Print gap
    print(" " * GAP, end="")

    # Print right pyramid
    # Print hashes
    print("#" * (i + 1), end="")

    # Print new line
    print("")

File: pset1/mario/more/mario.c
// ------------------------------------------ //
// CS50 Problem Set 1: Mario (more)
//
// Summary: Prints Mario`s pyramid (more)
//
// Lucas Emidio Fernandes Dias
// 9 March 2018
// ------------------------------------------ //

// Include libraries
#include <stdio.h>
#include <cs50.h>

// Constants
#define GAP 2 // Gap width

// ------------------------------------------ //
//        MAIN                                //
// ------------------------------------------ //
int main(void)
{
    int height; // Height of the pyramid (user input)

    // Asks the user for a height
    do
    {
        height = get_int("Height: ");
    }
    // Height is only valid if 0 <= height <= 23
    while (height < 0 || height > 23);

    // Prints the pyramid
    for (int i = 0; i < height; i++)
    {
        // Prints left pyramid
        // Print spaces
        for (int j = 0; j < height - i - 1; j++)
        {
            printf(" ");
        }
        // Print hashes
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }

        // Print gap
        for (int j = 0; j < GAP; j++)
        {
            printf(" ");
        }

        // Print right pyramid
        // Print hashes
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }

        // Print new line
        printf("\n");
    }
}